### PR TITLE
obs-plugin-droidcam: Add version 1.6.0

### LINKED ISSUE #7107: droidcamobs Request
PackageIdentifier: Droidcamobs
PackageVersion: 1.3
PackageLocale: en-US
Publisher: droidcamobs
PackageName: droidcamobs
License: open source
ShortDescription: webcam emulator between smartphone and pc
Installers:

Architecture: x64
InstallerType: exe
InstallerUrl: https://files.dev47apps.net/obs/DroidCamOBS.Setup.1.3.exe
ManifestType: singleton
ManifestVersion: 1.0.0

### PR DESCRIPTION
**Pull request by Zliced13**
Closes  #7107

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Jimimaku** [comment]
> Alle Änderungen sehen gut aus.
> 
> Warten Sie auf die Überprüfung durch menschliche Mitarbeiter.
> 
> ### droidcam-obs-plugin
> * [x] Beschreibung
> * [x] Lizenz
> * [x]  Hashes
> * [x] Schachbrett
> * [x] Automatische Aktualisierung

**Comment by Jimimaku** [comment]
> Closes #7107
> 
> * [x]  I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

**Comment by HUMORCE** [comment]
for plugins of other apps: `<app>-plugin-<plugin_name>.json`

```
❯ scoop search obs
Results from local buckets...

Name             Version      Source   Binaries
----             -------      ------   --------
obs-studio-small 21.1.2       extras
obs-studio       28.1.2       extras
streamlabs-obs   1.12.1       extras
obs-studio-pre   29.0.0-beta1 versions
obs-studio27     27.2.4       versions
```

consider the variant versions of OBS, `obs-plugin-droidcam` is acceptable.

here is an example:
```
❯ scoop search keepass
Results from local buckets...

Name                               Version Source   Binaries
----                               ------- ------   --------
keepass-plugin-keeagent            0.13.3  extras
keepass-plugin-keeanywhere         2.0.3   extras
keepass-plugin-keepasshttp         1.8.4.2 extras
keepass-plugin-keepassnatmsg       2.0.14  extras
keepass-plugin-keepassrpc          1.14.0  extras
keepass-plugin-keetraytotp         0.108.0 extras
keepass-plugin-kpenhentryview      2.4.3   extras
keepass-plugin-readable-passphrase 1.3.0   extras
keepass-plugin-sequencer           0.1.1   extras
keepass-plugin-webautotype         6.8.2   extras
keepass-plugin-yafd                1.2.5.0 extras
keepass                            2.52    extras
keepassxc                          2.7.4   extras
keepassxc2.6.6                     2.6.6   versions
```

File: bucket/obs-plugin-droidcam.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        }
+    },
+    "extract_dir": "$_13_",
+    "pre_install": [
+        "Remove-Item \"$dir\\data\\obs-plugins\\droidcam-obs\\`$_13_\" -Recurse",
+        "if (Test-Path \"$(appdir obs-studio $global)\") {",
+        "   Copy-Item \"$dir\\data\\obs-plugins\\droidcam-obs\" \"$(appdir obs-studio $global)\\current\\data\\droidcam-obs\" -Recurse -ErrorAction 'SilentlyContinue'",
+        "   Copy-Item \"$dir\\obs-plugins\\64bit\\*\" \"$(appdir obs-studio $global)\\current\\obs-plugins\\64bit\" -Recurse -ErrorAction 'SilentlyContinue'",
+        "}",
+        "elseif (!(Test-Path \"$(appdir obs-studio $global)\")) {error \"$app requires 'obs-studio' to be installed\"; break}"
```
Comment (HUMORCE) [inline bucket/obs-plugin-droidcam.json]: we should use `depends`, even if does not consider other variant versions. but we are considering them now, so just drop this line.

**Review by HUMORCE — CHANGES_REQUESTED** [review]

File: bucket/obs-plugin-droidcam.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{
+    "version": "1.6.0",
+    "homepage": "https://www.dev47apps.com/obs/",
+    "description": "An OBS plugin that allows the user to use their phone as a webcam for OBS Studio.",
+    "license": "GPL-2.0",
+    "architecture": {
```
Comment (HUMORCE) [inline bucket/obs-plugin-droidcam.json:6]:
```suggestion
    "license": "GPL-2.0",
    "suggest": {
        "OBS": "obs-studio"
    },
    "architecture": {
```

File: bucket/obs-plugin-droidcam.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        "elseif (Test-Path \"$(appdir obs-studio27 $global)\") {",
+        "   Copy-Item \"$dir\\data\\obs-plugins\\droidcam-obs\" \"$(appdir obs-studio27 $global)\\current\\data\\droidcam-obs\" -Recurse -ErrorAction 'SilentlyContinue'",
+        "   Copy-Item \"$dir\\obs-plugins\\64bit\\*\" \"$(appdir obs-studio27 $global)\\current\\obs-plugins\\64bit\" -Recurse -ErrorAction 'SilentlyContinue'",
+        "}",
+        "elseif (Test-Path \"$(appdir obs-studio-pre $global)\") {",
+        "   Copy-Item \"$dir\\data\\obs-plugins\\droidcam-obs\" \"$(appdir obs-studio-pre $global)\\current\\data\\droidcam-obs\" -Recurse -ErrorAction 'SilentlyContinue'",
+        "   Copy-Item \"$dir\\obs-plugins\\64bit\\*\" \"$(appdir obs-studio-pre $global)\\current\\obs-plugins\\64bit\" -Recurse -ErrorAction 'SilentlyContinue'",
+        "}",
+        "else { error \"$app requires any obs-studio variant to be installed!`nPlease install either,`nextras/obs-studio`nextras/obs-studio-small`nversions/obs-studio27`nversions/obs-studio-pre\"; break }"
+    ],
```
Comment (HUMORCE) [inline bucket/obs-plugin-droidcam.json:32]:
```suggestion
    "pre_install": [
        "Remove-Item \"$dir\\data\\obs-plugins\\droidcam-obs\\`$_13_\" -Recurse",
        "'obs-studio', 'obs-studio-small', 'obs-studio27', 'obs-studio-pre' | ForEach-Object {",
        "    $obs = \"$(appdir $_ $global)\"",
        "    if (Test-Path $obs) {",
        "        info \"Install for $_\"",
        "        Copy-Item \"$dir\\data\" \"$obs\\current\" -Recurse -ErrorAction SilentlyContinue",
        "        Copy-Item \"$dir\\obs-plugins\" \"$obs\\current\" -Recurse -ErrorAction SilentlyContinue",
        "    }",
        "}"
    ],
```

File: bucket/obs-plugin-droidcam.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        "elseif (Test-Path \"$(appdir obs-studio27 $global)\") {",
+        "   Remove-Item \"$(appdir obs-studio27 $global)\\current\\data\\droidcam-obs\" -Recurse -ErrorAction 'SilentlyContinue' -Force",
+        "   Remove-Item \"$(appdir obs-studio27 $global)\\current\\obs-plugins\\64bit\\droidcam-obs.*\" -Recurse -ErrorAction 'SilentlyContinue' -Force",
+        "}",
+        "elseif (Test-Path \"$(appdir obs-studio-pre $global)\") {",
+        "   Remove-Item \"$(appdir obs-studio-pre $global)\\current\\data\\droidcam-obs\" -Recurse -ErrorAction 'SilentlyContinue' -Force",
+        "   Remove-Item \"$(appdir obs-studio-pre $global)\\current\\obs-plugins\\64bit\\droidcam-obs.*\" -Recurse -ErrorAction 'SilentlyContinue' -Force",
+        "}",
+        "else { continue }"
+    ],
```
Comment (HUMORCE) [inline bucket/obs-plugin-droidcam.json:51]:
```suggestion
    "pre_uninstall": [
        "'obs-studio', 'obs-studio-small', 'obs-studio27', 'obs-studio-pre' | ForEach-Object {",
        "    $obs = \"$(appdir $_ $global)\"",
        "    if (Test-Path $obs) {",
        "        info \"Uninstall for $_\"",
        "        Remove-Item \"$obs\\current\\data\\droidcam-obs\" -Force -Recurse -ErrorAction SilentlyContinue",
        "        Remove-Item \"$obs\\current\\obs-plugins\\64bit\\droidcam-obs.*\" -Force -ErrorAction SilentlyContinue",
        "    }",
        "}"
    ],
```

**Review by HUMORCE — CHANGES_REQUESTED** [review]

**Comment by Zliced13** [comment]
@HUMORCE It's important to keep the `else` function, or it will not work properly.

**Comment by Zliced13** [comment]
@HUMORCE don't merge, I am still running some tests of the manifest on my machine. I will let you know when this PR is ready.

**Comment by Zliced13** [comment]
The current manifest is not functioning properly. Why does **Scoop** only execute the first command in the `if` statement and skips the other commands? It then executes the `elseif` statement for some stupid reason, which causes a install failure.
(VSCodium)n5J8M4_11-27-2022

```
> scoop install obs-plugin-droidcam                                                                   18:31:53
Installing 'obs-plugin-droidcam' (1.6.0) [64bit] from MyScoop bucket
Loading DroidCamOBS.Setup.1.6.0.exe from cache.
Checking hash of DroidCamOBS.Setup.1.6.0.exe ... ok.
Extracting dl.7z ... done.
Running pre_install script...
INFO  Install for obs-studio Copy-Item
ERROR obs-plugin-droidcam requires any obs-studio variant to be installed!
Please install either,
extras/obs-studio
extras/obs-studio-small
versions/obs-studio27
versions/obs-studio-pre
```

(WindowsTerminal)G436C0_11-27-2022

**Comment by HUMORCE** [comment]
this plugin is not a development tool or one of development toolchain, we should not use method such as `depends` to restrict installation.

expected behaviour: decompress file and nothing happens, if OBS not found.

**Comment by Zliced13** [comment]
> this plugin is not a development tool or one of development toolchain, we should not use method such as `depends` to restrict installation.
> 
> expected behaviour: decompress file and nothing happens, if OBS not found.

I'm not using the `depends` property in my manifest.

File: bucket/obs-plugin-droidcam.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        "Remove-Item \"$dir\\data\\obs-plugins\\droidcam-obs\\`$_13_\" -Recurse",
+        "'obs-studio', 'obs-studio-small', 'obs-studio27', 'obs-studio-pre' | ForEach-Object {",
+        "   $obs = \"$(appdir @($_) $global)\"",
+        "   if (Test-Path \"$obs\") {",
+        "       info \"Install for $_\"",
+        "       Copy-Item \"$dir\\data\\obs-plugins\\droidcam-obs\" \"$obs\\current\\data\\obs-plugins\" -Recurse -ErrorAction SilentlyContinue",
+        "       Copy-Item \"$dir\\obs-plugins\\64bit\\*\" \"$obs\\current\\obs-plugins\\64bit\" -Recurse -ErrorAction SilentlyContinue",
+        "   }",
+        "}"
+    ],
```
Comment (HUMORCE) [inline bucket/obs-plugin-droidcam.json:32]:
```suggestion
    "pre_install": [
        "Remove-Item \"$dir\\data\\obs-plugins\\droidcam-obs\\`$_13_\" -Recurse",
        "'obs-studio', 'obs-studio-small', 'obs-studio27', 'obs-studio-pre' | ForEach-Object {",
        "    $obs = \"$(appdir $_ $global)\"",
        "    if (Test-Path $obs) {",
        "       info \"Install for $_\"",
        "       Copy-Item \"$dir\\data\" \"$obs\\current\" -Recurse -ErrorAction SilentlyContinue",
        "       Copy-Item \"$dir\\obs-plugins\" \"$obs\\current\" -Recurse -ErrorAction SilentlyContinue",
        "   }",
        "}"
    ],
```

Comment (HUMORCE) [reply bucket/obs-plugin-droidcam.json:32]: styles L25-26 (L35-36)
shortens L28-29 (there are no compatibility issues with windows powershell 5)

**Review by HUMORCE — CHANGES_REQUESTED** [review]

**Review by HUMORCE — APPROVED** [review]

**Pull request merged**